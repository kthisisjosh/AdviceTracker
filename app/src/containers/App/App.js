import React from "react"
import { BrowserRouter as Router, Switch, Route } from "react-router-dom"
import DashboardPage from "../DashboardPage/DashboardPage"
import LoginPage from "../LoginPage/LoginPage"
import LandingPage from "../LandingPage/LandingPage"
import ProfilePage from "../ProfilePage/ProfilePage"
import BrowsePage from "../BrowsePage/BrowsePage"
import SubmitPage from "../SubmitPage/SubmitPage"
import CategoryPage from "../CategoryPage/CategoryPage"
import SubCategoryPage from "../SubCategoryPage/SubCategoryPage"
import AdvicePage from "../AdvicePage/AdvicePage"
import "../../app.css"

//Redux
import { Provider } from "react-redux"
import store from "../../redux"

function App() {
    return (
        <Provider store={store}>
            <Router>
                <Switch>
                    <Route component={LandingPage} exact path="/" />
                    <Route component={LoginPage} exact path="/login" />
                    <Route component={DashboardPage} exact path="/dashboard" />
                    <Route exact path="/dashboard/category/:id" component={CategoryPage} />
                    <Route exact path="/dashboard/subcategory/:id" component={SubCategoryPage} />
                    <Route exact path="/dashboard/advice/:id" component={AdvicePage} />
                    <Route component={ProfilePage} exact path="/user/:username" />
                    <Route component={BrowsePage} exact path="/browse" />
                    <Route component={null} exact path="/browse/:category/:content" />
                    <Route component={SubmitPage} exact path="/submit" />
                </Switch>
            </Router>
        </Provider>
    )
}

export default App
