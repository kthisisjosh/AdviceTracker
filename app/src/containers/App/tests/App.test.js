import React from "react"
import App from "../App"
import ShallowRenderer from "react-test-renderer/shallow"

const renderer = new ShallowRenderer()

describe("<App />", () => {
    it("should render and match the snapshot", () => {
        renderer.render(<App />)
        const renderedOutput = renderer.getRenderOutput()
        expect(renderedOutput).toMatchSnapshot()
    })
})

