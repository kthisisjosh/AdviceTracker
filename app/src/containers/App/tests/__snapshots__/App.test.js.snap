// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(LoginPage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/login"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(DashboardPage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/dashboard"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(CategoryPage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/dashboard/category/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(SubCategoryPage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/dashboard/subcategory/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AdvicePage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/dashboard/advice/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(ProfilePage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/user/:username"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(BrowsePage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/browse"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(PostPage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/browse/:category/:content"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(SubmitPage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/submit"
      />
    </Switch>
  </BrowserRouter>
</Provider>
`;
